include/openenclave/internal/sgx/plugin.h

```diff
@@ -5,87 +5,18 @@
+#include <openenclave/internal/crypto/sha.h>
 
 OE_EXTERNC_BEGIN
 
-/**
- * The SGX plugin UUID.
- */
-#define OE_SGX_PLUGIN_UUID                                                \
-    {                                                                     \
-        0xa3, 0xa2, 0x1e, 0x87, 0x1b, 0x4d, 0x40, 0x14, 0xb7, 0x0a, 0xa1, \
-            0x25, 0xd2, 0xfb, 0xcd, 0x8c                                  \
-    }
-
-#define OE_FORMAT_UUID_SGX_ECDSA_P256 OE_SGX_PLUGIN_UUID
-
-#define OE_FORMAT_UUID_SGX_LOCAL_ATTESTATION                              \
-    {                                                                     \
-        0x09, 0x26, 0x8c, 0x33, 0x6e, 0x0b, 0x45, 0xe5, 0x8a, 0x27, 0x15, \
-            0x64, 0x4d, 0x0e, 0xf8, 0x9a                                  \
-    }
-
-#define OE_FORMAT_UUID_SGX_EPID_LINKABLE                                  \
-    {                                                                     \
-        0xf2, 0x28, 0xaa, 0x3f, 0xde, 0x4d, 0x49, 0xd3, 0x88, 0x4c, 0xb2, \
-            0xaa, 0x87, 0xa5, 0x0d, 0xa6                                  \
-    }
-
-#define OE_FORMAT_UUID_SGX_EPID_UNLINKABLE                                \
-    {                                                                     \
-        0x5c, 0x35, 0xd2, 0x90, 0xa2, 0xc2, 0x4c, 0x55, 0x9e, 0x13, 0x5a, \
-            0xd7, 0x32, 0x74, 0x6c, 0x88                                  \
-    }
-
-#define OE_SGX_PLUGIN_CLAIMS_VERSION 1
-
-/**
- *  Serialized header for the custom claims.
- */
-typedef struct _oe_sgx_plugin_claims_header
+typedef enum _sgx_evidence_format_type_t
 {
-    uint64_t version;
-    uint64_t num_claims;
-} oe_sgx_plugin_claims_header_t;
-
-/**
- * Serialzied entry for custom claims. Each entry will have the name and value
- * sizes and then the contents of the name and value respectively.
- */
-typedef struct _oe_sgx_plugin_claims_entry
-{
-    uint64_t name_size;
-    uint64_t value_size;
-    uint8_t name[];
-    // name_size bytes follow.
-    // value_size_bytes follow.
-} oe_sgx_plugin_claims_entry_t;
-
-/**
- * oe_sgx_serialize_claims
- *
- * Serializes a list of claims.
- *
- * This is available in the enclave only.
- *
- * @experimental
- *
- * @param [in] custom_claims Custom claims to serialize.
- * @param [in] custom_claims_length Length of custom_claims.
- * @param [out] claims_out Pointer to the address of a dynamically
- * allocated buffer holding the serialized custom claims.
- * @param [out] claims_size_out Size of the serialized custom claims.
- * @retval OE_OK on success.
- * @retval OE_INVALID_PARAMETER At least one parameter is invalid.
- * @retval An appropriate error code on failure.
- */
-struct _OE_SHA256;
-oe_result_t oe_sgx_serialize_claims(
-    const oe_claim_t* custom_claims,
-    size_t custom_claims_length,
-    uint8_t** claims_out,
-    size_t* claims_size_out,
-    struct _OE_SHA256* hash_out);
+    SGX_FORMAT_TYPE_LOCAL = 1,         // Evidence for SGX local attestation
```

Comment:
My understanding is:
1 is generated by oe_get_evidence(LOCAL_UUID)
2 is generated by oe_get_evidence(REMOTE_UUID)
3 is generated by oe_get_report(REMOTE)
4 is generated by SGX quote generator

Is that correct?

Reply:
Yes.